lms/djangoapps/api_manager/management/commands/bulk_delete_courses_with_reference_data.py

```diff
@@ -0,0 +1,110 @@
+"""
+import pytz
+import logging
+from optparse import make_option
+from datetime import datetime, timedelta
+from util.prompt import query_yes_no
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+from api_manager.models import CourseGroupRelationship, CourseContentGroupRelationship
+from openedx.core.djangoapps.course_groups.models import CourseCohortsSettings, CourseUserGroup
+from openedx.core.djangoapps.content.course_metadata.models import CourseAggregatedMetaData
+from openedx.core.djangoapps.content.course_structures.models import CourseStructure
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from courseware.models import StudentModule
+from progress.models import CourseModuleCompletion, StudentProgress, StudentProgressHistory
+from gradebook.models import StudentGradebook, StudentGradebookHistory
+from student.models import CourseAccessRole, CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Deletes courses based on their age and relevant data of each course
+    """
+    help = 'Deletes courses based on their age and relevant data of each course'
+    option_list = BaseCommand.option_list + (
+        make_option(
+            "-c",
+            "--age",
+            dest="age",
+            help="Age in days",
+            metavar="60"
+        ),
+    )
+    module_store = modulestore()
+    total_courses = 0
+    total_deleted = 0
+
+    @staticmethod
+    def delete_reference_data(course_key):
+        """
+        Deletes reference data for course
+        """
+        print 'removing reference data for course %s' % unicode(course_key)
+        CourseGroupRelationship.objects.filter(course_id=course_key).delete()
+        CourseContentGroupRelationship.objects.filter(course_id=course_key).delete()
+        CourseCohortsSettings.objects.filter(course_id=course_key).delete()
+        CourseUserGroup.objects.filter(course_id=course_key).delete()
+        CourseAggregatedMetaData.objects.filter(id=course_key).delete()
+        CourseStructure.objects.filter(course_id=course_key).delete()
+        CourseOverview.objects.filter(id=course_key).delete()
+        StudentModule.objects.filter(course_id=course_key).delete()
+        CourseModuleCompletion.objects.filter(course_id=course_key).delete()
+        StudentProgressHistory.objects.filter(course_id=course_key).delete()
+        StudentProgress.objects.filter(course_id=course_key).delete()
+        StudentGradebookHistory.objects.filter(course_id=course_key).delete()
+        StudentGradebook.objects.filter(course_id=course_key).delete()
+        CourseAccessRole.objects.filter(course_id=course_key).delete()
+        CourseEnrollment.objects.filter(course_id=course_key).delete()
```

Comment:
I would be interested to know how you compiled this set of references. Seems like a painstaking task! Are you confident this is comprehensive for the solutions fork? I would love to compile a similar list for edx-platform proper at some point.